src/main_targil3.js

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+
+
+function Page1(props) {
+    const { setPage, userName, setUserName, pass, setPass } = props;
+
+    return (
+        <html>
+
+            <head>
+                <title>Page 1</title>
+            </head>
+
+            <body>
+                <div>
+                    <lable>User name: </lable>
+                    <input type="text" value={userName} onChange={e => setUserName(e.target.value)}></input>
+                </div>
+                <div>
+                    <lable>Password: </lable>
+                    <input type="password" value={pass} onChange={e => setPass(e.target.value)}></input>
+                </div>
+                <div>
+                    <button onClick={e => setPage(1)}>Next</button>
```

Comment:
The page changing functionality should be outside of the page components, they do not need to be aware of the page change in any way.